pkg/cloudproviders/aws/controllers/interruption/events.go

```diff
@@ -0,0 +1,193 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package interruption
+
+import (
+	"fmt"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/aws/karpenter/pkg/cloudproviders/aws/events"
+)
+
+var _ events.Event = EC2SpotInterruptionWarning{}
+
+type EC2SpotInterruptionWarning struct {
```

Comment:
Including Type() in the type name feels a bit leaky. WDYT:
```suggestion
type EC2SpotInterruption struct {
```